src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -302,8 +302,8 @@ private PackageSource GetPackageSource(string key, Dictionary<string, IndexedPac
 
                 foreach (var packageSource in _configurationDefaultSources)
                 {
-                    var isSourceMatch = loadedPackageSources.Any(p => p.Source.Equals(packageSource.Source, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Related to https://github.com/NuGet/NuGet.Client/pull/4264/files#r708797119
Triggered for `nuget sources Add -Name "MyServer" -Source \\myserver\packages`.
Looks not really relevant anymore.